src/BTVNanoCommissioning/workflows/QCD_validation.py

```diff
@@ -287,31 +351,41 @@ def process_shift(self, events, shift_name):
                         )
 
             output["njet"].fill(syst, njet, weight=weight)
-            if "JetSVs" in events.Jet.fields:
-                output["nJetSVs"].fill(syst, nJetSVs, weight=weight)
-
-                output["dr_SVjet0"].fill(
-                    syst,
-                    flatten(lj_matched_JetSVs_genflav),
-                    flatten(abs(lj_SVs.deltaR) - 0.1),
-                    weight=flatten(
-                        ak.broadcast_arrays(weight, lj_matched_JetSVs["pt"])[0]
-                    ),
-                )
+            if "JetSVs" in events.fields:
+                if len(lj_matched_JetSVs) > 0:
+                    output["nJetSVs"].fill(syst, nJetSVs, weight=weight[valid_events])
+                    output["dr_SVjet0"].fill(
+                        syst,
+                        flatten(lj_matched_JetSVs_genflav),
+                        flatten(abs(lj_SVs.deltaR) - 0.1),
+                        weight=flatten(
+                            ak.broadcast_arrays(
+                                weight[valid_events], lj_matched_JetSVs["pt"]
+                            )[0]
+                        ),
+                    )
             output["npvs"].fill(syst, flatten(selev.PV.npvsGood), weight=weight)
             if not isRealData:
                 if syst == "nominal":
-                    output["pu"].fill(
-                        "noPU",
-                        flatten(selev.Pileup.nTrueInt),
-                        weight=np.ones_like(weight),
-                    )
-                    output["npvs"].fill(
-                        "noPU",
-                        flatten(selev.Pileup.nTrueInt),
-                        weight=np.ones_like(weight),
-                    )
+                    # output["pu"].fill(
+                    #    "noPU",
+                    #    flatten(selev.Pileup.nTrueInt),
+                    #    weight=np.ones_like(weight),
+                    # )
+                    # output["npvs"].fill(
+                    #    "noPU",
+                    #    flatten(selev.Pileup.nTrueInt),
+                    #    weight=np.ones_like(weight),
+                    # )
                     output["pu"].fill(syst, flatten(selev.PV.npvsGood), weight=weight)
```

Comment:
can you please completely remove the lines here

Reply:
Removed